### PR TITLE
feat(transformer/react): read comment pragma @jsxRuntime classic / automatic

### PR DESCRIPTION
**Pull request by Boshen**
closes #1120

### CODE REVIEW

**Comment by Boshen** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1133** <a href="https://app.graphite.dev/github/pr/web-infra-dev/oxc/1133" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/web-infra-dev/oxc/1133?utm_source=stack-comment).

**Comment by Boshen** [comment]
@Dunqing Can you help out and review as well.

File: crates/oxc_transformer/src/react_jsx/options.rs [hunk 1]
```diff
@@ -28,3 +30,30 @@ impl ReactJsxRuntime {
...
+    pub(crate) fn with_comments(mut self, semantic: &Semantic) -> Self {
+        for (_, span) in semantic.trivias().comments_spans() {
+            let comment = span.source_text(semantic.source_text());
+            // strip leading jsdoc comment `*` and then whitespaces
+            let comment = comment.strip_prefix('*').unwrap_or(comment).trim_start();
+            // strip leading `@`
+            let Some(comment) = comment.strip_prefix('@') else { continue };
+            // read jsxRuntime
+            match comment.strip_prefix("jsxRuntime").map(str::trim) {
+                Some("classic") => self.runtime = ReactJsxRuntime::Classic,
```
Comment (Dunqing) [inline crates/oxc_transformer/src/react_jsx/options.rs:52]: Do we need to store every option in a struct? Can it be stored in self.options.xxx? We still have a lot of options that we don't support yet, and I'm kind of curious about that

Comment (Boshen) [reply crates/oxc_transformer/src/react_jsx/options.rs:52]: I don't get your question, this is saving the value into `ReactJsxOptions`.

Comment (Dunqing) [reply crates/oxc_transformer/src/react_jsx/options.rs:52]: Ah. I see, Thank you!

**Pull request merged**